### PR TITLE
ElementR: Process all verification events, not just requests

### PR DESCRIPTION
**Pull request by florianduros**
## Checklist

-   [x] Tests written for new code (and old code if feasible)
-   [x] Linter and other CI checks pass
-   [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-js-sdk/blob/develop/CONTRIBUTING.md))

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * ElementR: Process all verification events, not just requests ([\#3650](https://github.com/matrix-org/matrix-js-sdk/pull/3650)). Contributed by @florianduros.

### CODE REVIEW

File: src/rust-crypto/rust-crypto.ts [hunk 1]
```diff
@@ -1040,15 +1040,12 @@ export class RustCrypto extends TypedEventEmitter<RustCryptoEvents, RustCryptoEv
      * @param event - live event
      */
     public async onLiveEventFromSync(event: MatrixEvent): Promise<void> {
-        // Ignore state event
-        if (event.isState()) return;
+        // Ignore state event or remote echo
+        if (event.isState() || event.getUnsigned().transaction_id === null) return;
```
Comment (richvdh) [inline src/rust-crypto/rust-crypto.ts:1044]: why would a remote echo have a transaction ID of `null` ?

File: src/rust-crypto/verification.ts [hunk 2]
```diff
@@ -700,3 +702,31 @@ export function verificationMethodIdentifierToMethod(method: string): RustSdkCry
...
+    switch (event.getType()) {
+        case EventType.KeyVerificationCancel:
+        case EventType.KeyVerificationDone:
+        case EventType.KeyVerificationMac:
+        case EventType.KeyVerificationStart:
+        case EventType.KeyVerificationKey:
+        case EventType.KeyVerificationReady:
+        case EventType.KeyVerificationAccept: {
+            return true;
+        }
```
Comment (richvdh) [inline src/rust-crypto/verification.ts:724]: are the curly braces here (and below) required by our linter? I find them a bit distracting.

```suggestion
        case EventType.KeyVerificationAccept:
            return true;

```

File: src/rust-crypto/verification.ts [hunk 3]
```diff
@@ -700,3 +702,31 @@ export function verificationMethodIdentifierToMethod(method: string): RustSdkCry
     }
     return meth;
 }
+
+/**
+ * Return true if the event is a verification event
```
Comment (richvdh) [inline src/rust-crypto/verification.ts:707]: Let's be a bit more specific about what we are matching here.

```suggestion
 * Return true if the event's type matches that of an in-room verification event
```

File: src/rust-crypto/rust-crypto.ts (hunk 1 above)
Comment (florianduros) [reply src/rust-crypto/rust-crypto.ts]: I changed, it was a misunderstanding of mine

File: src/rust-crypto/rust-crypto.ts [hunk 4]
```diff
@@ -1041,7 +1041,8 @@ export class RustCrypto extends TypedEventEmitter<RustCryptoEvents, RustCryptoEv
      */
     public async onLiveEventFromSync(event: MatrixEvent): Promise<void> {
         // Ignore state event or remote echo
-        if (event.isState() || event.getUnsigned().transaction_id === null) return;
+        // transaction_id is provided in case of remote echo {@link https://spec.matrix.org/v1.7/client-server-api/#local-echo}
+        if (event.isState() || "transaction_id" in event.getUnsigned()) return;
```
Comment (richvdh) [inline src/rust-crypto/rust-crypto.ts:1045]: although this is technically correct, in other places where we make this check, we look for truthiness of `transaction_id`. In other words:

```suggestion
        if (event.isState() || !!event.getUnsigned().transaction_id) return;
```

Let's be consistent.

**Review by richvdh — APPROVED** [review]
lgtm otherwise

**Pull request merged**